qpmodel/SQLite.g4

```diff
@@ -162,34 +162,62 @@ column_constraint
     AND
     OR
 */
-expr
- : literal_value											#LiteralExpr
- | BIND_PARAMETER											#bindexpr
- | ( ( database_name '.' )? table_name '.' )? column_name	#ColExpr
- | unary_operator expr										#unaryexpr
- | expr '||' expr											#strconexpr
- | expr op=( '*' | '/' | '%' ) expr							#arithtimesexpr
- | expr op=( PLUS | MINUS ) expr							#arithplusexpr
- | expr op=( '<<' | '>>' | '&' | '|' ) expr					#arithbitexpr
- | expr op=( '<' | '<=' | '>' | '>=' ) expr					#arithcompexpr																		
- | expr K_NOT? K_BETWEEN '(' expr ',' expr ')'				#BetweenExpr
- | expr K_IS K_NOT? expr									#IsExpr
- | expr K_NOT? K_LIKE expr									#LikeExpr
- | expr op=( '=' | '==' | '!=' | '<>' 
-	 | K_GLOB | K_MATCH | K_REGEXP ) expr	                #BoolEqualexpr
- | expr K_NOT? K_IN ( '(' ( select_stmt						
-                          | expr ( ',' expr )*
+logical_expr
+ : logical_expr K_AND logical_expr											#LogicAndExpr
+ | logical_expr K_OR logical_expr											#LogicOrExpr
+ | bool_expr                                                         #boolexpr
+ | '(' logical_expr ')'												#brackexpr
+;
+
+bool_expr
```

Comment:
rename as predicate_expr? because bool_expr shall also contain logic_expr?

Reply:
> should I separate expr into logical_expr, bool_expr and arith_expr, I don't have theory support but just lots of tries.

I don't think there is a definite rule how they shall be splitted. Current one looks good to me.

Reply:
what is three ( .. )?